Ginger/Ginger/SourceControl/SourceControlConnDetailsPage.xaml.cs

```diff
@@ -59,13 +59,27 @@ private void Bind()
             SourceControlClassTextBox.IsReadOnly = true;
             SourceControlClassTextBox.IsEnabled = false;
 
-            SourceControlURLTextBox.Text = SourceControlIntegration.GetRepositoryURL( App.UserProfile.Solution.SourceControl);
-                SourceControlURLTextBox.IsReadOnly = true;
-                SourceControlURLTextBox.IsEnabled = false;
-
-                App.ObjFieldBinding(SourceControlUserTextBox, TextBox.TextProperty, App.UserProfile.Solution.SourceControl, SourceControlBase.Fields.SourceControlUser);
-                App.ObjFieldBinding(SourceControlPassTextBox, TextBox.TextProperty, App.UserProfile.Solution.SourceControl, SourceControlBase.Fields.SourceControlPass);
+            SourceControlURLTextBox.Text = SourceControlIntegration.GetRepositoryURL(App.UserProfile.Solution.SourceControl);
+            SourceControlURLTextBox.IsReadOnly = true;
+            SourceControlURLTextBox.IsEnabled = false;
 
+            App.ObjFieldBinding(SourceControlUserTextBox, TextBox.TextProperty, App.UserProfile.Solution.SourceControl, SourceControlBase.Fields.SourceControlUser);
+            App.ObjFieldBinding(SourceControlPassTextBox, TextBox.TextProperty, App.UserProfile.Solution.SourceControl, SourceControlBase.Fields.SourceControlPass);
+            if (SourceControlClassTextBox.Text == "GIT")
+            {
+                lblControlConnectionTimeout.Visibility = Visibility.Hidden;
```

Comment:
Put it in a stack panel and set the visibility for panel instead of setting for individual component